### PR TITLE
Source Iterable - better processing of 401 and 429 errors

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/829

## How
 - improve `discover`: return less streams if the api key does not provide access to all the streams
 - do not make calls for subsequent stream slices after getting 401 error because it likely leads to 429 error in other streams as well

### CODE REVIEW

**Review by arsenlosenko — APPROVED** [review]

File: airbyte-integrations/connectors/source-iterable/source_iterable/streams.py [hunk 1]
```diff
@@ -381,7 +394,7 @@ def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         if not self.check_unauthorized_key(response):
-            yield from []
+            return []
```
Comment (girarda) [inline airbyte-integrations/connectors/source-iterable/source_iterable/streams.py:397]: what's the benefit of returning `[]` instead of `yield from []`?

**Review by girarda — APPROVED** [review]

File: airbyte-integrations/connectors/source-iterable/source_iterable/streams.py (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-iterable/source_iterable/streams.py:397]: after yielding from the empty list generator will execute the following code and yield `records` as well whereas `return` will return the empty list once and finish executing the routine

**Comment by dancook-doxo** [comment]
I upgraded Airbyte OS to Iterable 0.1.20 (from 0.1.19) and got this error upon any attempt to sync:
2022-10-26 08:46:30 - Additional Failure Information: "The requested stream users was not found in the source. Available streams: dict_keys(['campaigns', 'campaigns_metrics', 'channels', 'lists', 'message_types', 'metadata', 'templates'])"

When I reverted to 0.1.19 I was able to sync just fine with no other changes, indicating that the user stream is indeed available.

**Comment by davydov-d** [comment]
@dancook-doxo thanks for the feedback! I will take a look what could go wrong asap

**Comment by davydov-d** [comment]
@dancook-doxo please try the new version - 0.1.21

**Comment by dancook-doxo** [comment]
Using 0.1.21 I was able to sync just fine on one connection that _doesn't_ involve a `users` stream, even though that broke undewr 0.1.20. And I also synced fine on the connection which _does_ involve `users`, which also broke under 0.1.20.

**Comment by davydov-d** [comment]
Good, thanks for the quick response

**Pull request merged**